checker/jdk/nullness/src/org/w3c/dom/Document.java

```diff
@@ -0,0 +1,847 @@
+     * <br> DOM applications may use the
+     * <code>DOMImplementation.hasFeature(feature, version)</code> method
+     * with parameter values "XMLVersion" and "1.0" (respectively) to
+     * determine if an implementation supports [<a href='http://www.w3.org/TR/2004/REC-xml-20040204'>XML 1.0</a>]. DOM
+     * applications may use the same method with parameter values
+     * "XMLVersion" and "1.1" (respectively) to determine if an
+     * implementation supports [<a href='http://www.w3.org/TR/2004/REC-xml11-20040204/'>XML 1.1</a>]. In both
+     * cases, in order to support XML, an implementation must also support
+     * the "XML" feature defined in this specification. <code>Document</code>
+     *  objects supporting a version of the "XMLVersion" feature must not
+     * raise a <code>NOT_SUPPORTED_ERR</code> exception for the same version
+     * number when using <code>Document.xmlVersion</code>.
+     * @since DOM Level 3
+     */
+    @Pure public @Nullable String getXmlVersion();
+    /**
+     *  An attribute specifying, as part of the <a href='http://www.w3.org/TR/2004/REC-xml-20040204#NT-XMLDecl'>XML declaration</a>, the version number of this document. If there is no declaration and if
+     * this document supports the "XML" feature, the value is
+     * <code>"1.0"</code>. If this document does not support the "XML"
+     * feature, the value is always <code>null</code>. Changing this
+     * attribute will affect methods that check for invalid characters in
+     * XML names. Application should invoke
+     * <code>Document.normalizeDocument()</code> in order to check for
+     * invalid characters in the <code>Node</code>s that are already part of
+     * this <code>Document</code>.
+     * <br> DOM applications may use the
+     * <code>DOMImplementation.hasFeature(feature, version)</code> method
+     * with parameter values "XMLVersion" and "1.0" (respectively) to
+     * determine if an implementation supports [<a href='http://www.w3.org/TR/2004/REC-xml-20040204'>XML 1.0</a>]. DOM
+     * applications may use the same method with parameter values
+     * "XMLVersion" and "1.1" (respectively) to determine if an
+     * implementation supports [<a href='http://www.w3.org/TR/2004/REC-xml11-20040204/'>XML 1.1</a>]. In both
+     * cases, in order to support XML, an implementation must also support
+     * the "XML" feature defined in this specification. <code>Document</code>
+     *  objects supporting a version of the "XMLVersion" feature must not
+     * raise a <code>NOT_SUPPORTED_ERR</code> exception for the same version
+     * number when using <code>Document.xmlVersion</code>.
+     * @exception DOMException
+     *    NOT_SUPPORTED_ERR: Raised if the version is set to a value that is
+     *   not supported by this <code>Document</code> or if this document
+     *   does not support the "XML" feature.
+     * @since DOM Level 3
+     */
+    public void setXmlVersion(String xmlVersion)
```

Comment:
I think the documentation is misleading in this instance. The JavaDoc seems to be a translated version of the DOM specifications, which are somewhat more lenient than the actual JAXP implementations. Notably, the specifications assume that write and read operations are symmetric, whereas the JAXP implementations do not necessarily abide by such assumptions.

In this particular case, the default JAXP implementation bundled with the JVM will throw a `NullPointerException` if the argument is `null`. I'll add a comment here noting that. I'll also go through the other instances and check what behavior the default implementation exhibits.